test/integration/user_test.rb

```diff
@@ -12,4 +12,18 @@ class UserTest < ActionDispatch::IntegrationTest
     assert_submit_button(users_path)
     assert page.has_link? 'user12345'
   end
+
+  test "notice is removed after submit pressed" do
+    Capybara.current_driver = Capybara.javascript_driver
```

Comment:
Regarding putting it in the same file, you can use a `context` block to only set the javascript_driver for a particular set of tests, saves having it in each test then.